From e930c0a89fff7759e92457ea9306125ed99a0e89 Mon Sep 17 00:00:00 2001
From: Ivan Molodetskikh <yalterz@gmail.com>
Date: Fri, 9 Dec 2016 21:57:29 +0300
Subject: [PATCH 12/28] Don't sort points in interpolators, assume they're
 already sorted. This fixes bugs due to Sort not being a stable sort.

---
 osu!/Helpers/Interpolators/CubicSplineInterpolator.cs | 1 -
 osu!/Helpers/Interpolators/LinearInterpolator.cs      | 1 -
 osu!/Helpers/Interpolators/StreamInterpolator.cs      | 1 -
 3 files changed, 3 deletions(-)

diff --git a/osu!/Helpers/Interpolators/CubicSplineInterpolator.cs b/osu!/Helpers/Interpolators/CubicSplineInterpolator.cs
index f0547fe..6f758d1 100644
--- a/osu!/Helpers/Interpolators/CubicSplineInterpolator.cs
+++ b/osu!/Helpers/Interpolators/CubicSplineInterpolator.cs
@@ -50,7 +50,6 @@ namespace osu.Helpers.Interpolators
                 throw new ArgumentException("Cannot interpolate with less than 2 points.", "points");
 
             var points_ = new List<PointWithTime>(points);
-            points_.Sort();
 
             StartTime = points_[0].time;
             EndTime = points_[points_.Count - 1].time;
diff --git a/osu!/Helpers/Interpolators/LinearInterpolator.cs b/osu!/Helpers/Interpolators/LinearInterpolator.cs
index cc5486a..75fc4e1 100644
--- a/osu!/Helpers/Interpolators/LinearInterpolator.cs
+++ b/osu!/Helpers/Interpolators/LinearInterpolator.cs
@@ -17,7 +17,6 @@ namespace osu.Helpers.Interpolators
                 throw new ArgumentException("Cannot interpolate with less than 2 points.", "points");
 
             Points = new List<PointWithTime>(points);
-            Points.Sort();
 
             StartTime = Points[0].time;
             EndTime = Points[Points.Count - 1].time;
diff --git a/osu!/Helpers/Interpolators/StreamInterpolator.cs b/osu!/Helpers/Interpolators/StreamInterpolator.cs
index 3afa315..a5f16b5 100644
--- a/osu!/Helpers/Interpolators/StreamInterpolator.cs
+++ b/osu!/Helpers/Interpolators/StreamInterpolator.cs
@@ -19,7 +19,6 @@ namespace osu.Helpers.Interpolators
                 throw new ArgumentException("Cannot interpolate with less than 2 points.", "points");
 
             Points = new List<PointWithTime>(points);
-            Points.Sort();
 
             StartTime = Points[0].time;
             EndTime = Points[Points.Count - 1].time;
-- 
2.19.1

